test/performance/fixtures/metrics.js

```diff
@@ -0,0 +1,146 @@
+export class Metrics {
+	constructor( { page } ) {
+		this.page = page;
+	}
+
+	async getTimeToFirstByte() {
+		return await this.page.evaluate( () => {
+			// Based on https://web.dev/ttfb/#measure-ttfb-in-javascript
+			return new Promise( ( resolve ) => {
+				new PerformanceObserver( ( entryList ) => {
+					const [ pageNav ] =
+						entryList.getEntriesByType( 'navigation' );
+					resolve( pageNav.responseStart );
+				} ).observe( {
+					type: 'navigation',
+					buffered: true,
+				} );
+			} );
+		} );
+	}
+
+	async getLargestContentfulPaint() {
+		return await this.page.evaluate( () => {
+			// Based on https://www.checklyhq.com/learn/headless/basics-performance#largest-contentful-paint-api-largest-contentful-paint
+			return new Promise( ( resolve ) => {
+				new PerformanceObserver( ( entryList ) => {
+					const entries = entryList.getEntries();
+					// The last entry is the largest contentful paint.
+					const largestPaintEntry = entries.at( -1 );
+
+					resolve( largestPaintEntry.startTime );
+				} ).observe( {
+					type: 'largest-contentful-paint',
+					buffered: true,
+				} );
+			} );
+		} );
+	}
+
+	async getLoadingDurations() {
+		return await this.page.evaluate( () => {
+			const [
+				{
+					requestStart,
+					responseStart,
+					responseEnd,
+					domContentLoadedEventEnd,
+					loadEventEnd,
+				},
+			] = performance.getEntriesByType( 'navigation' );
+			const paintTimings = performance.getEntriesByType( 'paint' );
+
+			return {
+				// Server side metric.
+				serverResponse: responseStart - requestStart,
+				// For client side metrics, consider the end of the response (the
+				// browser receives the HTML) as the start time (0).
+				firstPaint:
+					paintTimings.find( ( { name } ) => name === 'first-paint' )
+						.startTime - responseEnd,
+				domContentLoaded: domContentLoadedEventEnd - responseEnd,
+				loaded: loadEventEnd - responseEnd,
+				firstContentfulPaint:
+					paintTimings.find(
+						( { name } ) => name === 'first-contentful-paint'
+					).startTime - responseEnd,
+				// This is evaluated right after Playwright found the block selector.
```

Comment:
Block selector? AFAICT this is not tied to blocks here, so perhaps remove the comment?

Reply:
Nice catch - that's a leftover from copypasting it from the tests. Thanks!

Reply:
Addressed in https://github.com/WordPress/gutenberg/pull/53808/commits/7000dec29793b8db426b80a4a70bdcdd823aa469